### ISSUE TITLE
Revision fingerprint checksum only matches exact filenames

### ISSUE THREAD

**Issue by boris-petrov** [ember-cli/ember-cli#6040]
If you have a file named `asd.js` and you use it inside an HTML file like so: `<script src="asd.js?some_query=some_value"></script>`, the name won't be fingerprinted inside the output HTML, but the file itself will be and this will blow up with a 404. This is probably not an issue in `ember-cli` but I'm not sure what you're using underneath so you could open a new issue in the relevant place.

Output from ember version --verbose:
```
ember-cli: 2.6.2
http_parser: 2.7.0
node: 6.2.2
v8: 5.0.71.52
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
os: linux x64
```

**Comment by nathanhammond** [comment]
You likely have an issue with your asset path. Files you include in your public directory end up in `/assets/(...)` If you're adding arbitrary JavaScript, however, you probably want to follow the directions here: https://ember-cli.com/user-guide/#javascript-assets

I'm closing this issue as I don't believe it to be a bug. If you believe you're still running into a bug please reopen the issue. :smile: If you need further support [come hang out with us on Slack](https://embercommunity.slack.com/archives/-ember-cli)!

**Comment by boris-petrov** [comment]
@nathanhammond - This is not about files imported with `app.import`. Let me give a little bit more details.

In my ember's `public` directory I have a static HTML file (something like `ember-app/public/some-file.html`) and a static JS file (something like `ember-app/public/asd.js`). In the HTML file there is a line like the following: `<script src="asd.js?some_query=some_value"></script>`. Normally `ember-cli` (or something) would pass through the JS files, fingerprint them and replace all occurrences to them in the static HTML files. However, if there is such a query parameter as in the example above (`some_query=some_value`), this doesn't work and in the HTML file the link remains to `asd.js` while the file has been renamed to something like `asd-4883737ebf8fb56c5bec516c8360f5e2.js`.

Is it more clear now?

**Comment by nathanhammond** [comment]
I appreciate the additional details. Here's the process it's going through:

1. https://github.com/rickharrison/broccoli-asset-rev => generate an asset map.
2. https://github.com/rickharrison/broccoli-asset-rewrite => parse HTML/CSS/JS files for assets in the asset map and rewrite them.

My suspicion is that `broccoli-asset-rewrite` isn't succeeding at matching the asset name with your additional querystring but I haven't tested this. Can you confirm if that is indeed the case and, if so, open an issue there?

As an aside: why do you need to append a static querystring to a static asset?

**Comment by boris-petrov** [comment]
You are correct, we narrowed the problem down to `broccoli-asset-rewrite` and found the issue. We'll create a pull-request in their project. I hope you can add the new version (if they release one) to ember-cli. :) Thanks!

For the side note - this is not our code, it is third-party code and we don't want to maintain a fork of it, so we just put it as it is. :)

**Comment by nathanhammond** [comment]
Once it's up please link that PR back to this thread so people can follow along. Thanks for your research and efforts! (That'll also help us make sure we pull it in once it lands.)

**Comment by boris-petrov** [comment]
The pull-request is this one:
https://github.com/rickharrison/broccoli-asset-rewrite/pull/40
I hope it is merged soon! Thanks!

**Comment by rickharrison** [comment]
This is now published as `broccoli-asset-rewrite@1.0.14`

**Comment by joankaradimov** [comment]
Thanks for the release!
